examples/two_dimensional_turbulence.jl

```diff
@@ -0,0 +1,58 @@
+# # Two dimensional turbulence example
+#
+# In this example, we initialize a random velocity field and observe its viscous,
+# turbulent decay in a two-dimensional domain.
+
+using Oceananigans, PyPlot, Statistics
+
+# In addition to importing plotting and statistics packages, we import
+# some types and functions from `Oceananigans` that will aid in the calculation
+# and visualization of voriticty.
+
+using Oceananigans: Face, Cell
+using Oceananigans.TurbulenceClosures: ∂x_faa, ∂y_afa
+
+# We instantiate the model with a simple isotropic diffusivity
+
+model = Model(
+        grid = RegularCartesianGrid(N=(128, 128, 1), L=(2π, 2π, 2π)),
+     closure = ConstantIsotropicDiffusivity(ν=1e-3, κ=1e-3)
+)
+
+# Our initial condition randomizes `u` and `v`. We also ensure that both have
+# zero mean for purely aesthetic reasons.
+
+ϵ(x, y, z) = rand()
+set!(model, u=ϵ, v=ϵ)
+
+model.velocities.u.data .-= mean(data(model.velocities.u))
+model.velocities.v.data .-= mean(data(model.velocities.v))
+
+# Next we define a function for calculating the vertical vorticity 
+# associated with the velocity fields `u` and `v`.
+
+function vorticity!(ω, u, v)
```

Comment:
Would it make sense to define a `vorticity!` diagnostic to avoid having to rely on unexported functions?

Reply:
Part of the purpose of this example was to demonstrate how a user might work with fields and define functions that act on fields. Defining a `vorticity!` diagnostic would subvert that intent.

Reply:
@ali-ramadhan your thoughts? Does my response resolve this comment?